docs/how-to-guides/running-feast-in-production.md

```diff
@@ -141,6 +202,146 @@ prediction = model.predict(feature_vector)
+The obvious drawback is that your service must be written on Python to use the Feast Python SDK. 
+A benefit to using a Python stack, though, is that you can enjoy production-grade services with integrations with many existing data science tools.
+
+To integrate online retrieval into your service use the following code:
+```python
+from feast import FeatureStore
+
+with open('feature_refs.json', 'r') as f:
+    feature_refs = json.loads(f)
+
+fs = FeatureStore(repo_path="production/")
+
+# Read online features
+feature_vector = fs.get_online_features(
+    feature_refs=feature_refs,
+    entity_rows=[{"driver_id": 1001}]
+).to_dict()
+```
+
+### 4.2. Consume features via HTTP API from Serverless Feature Server
+
+if you don't want to add the Feast Python SDK as a dependency, or your feature retrieval service is written in a non-Python language,
+Feast can deploy a simple feature server
+on serverless infrastructure (eg, AWS Lambda, Google Cloud Function) for you.
+This service will provide an HTTP API with JSON I/O, which can be easily used with any programming language.
+
+[Read more about this feature](../reference/alpha-aws-lambda-feature-server.md)
+
+### 4.3. Java based Feature Server deployed on Kubernetes
+
+For users with very latency-sensitive and high QPS use-cases, Feast offers a high-performance Java feature server.
+high performant Java implementation of feature server. Besides the benefits of running on JVM, this implementation also provides gRPC API, which guarantees good connection utilization and 
+small request / response body size (in comparison to JSON). 
+You will probably need Feast Java SDK to retrieve features from this service. Those SDKs will wrap all gRPC logic for you and instead will provide more convenient APIs. 
+
+The Java based feature server can be deployed to Kubernetes cluster via Helm charts in a few simple steps:
+
+1. Install [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) and [helm 3](https://helm.sh/)
+2. Add the Feast Helm repository and download the latest charts:
+```
+helm repo add feast-charts https://feast-helm-charts.storage.googleapis.com
+helm repo update
+```
+3. Run Helm Install
+```
+helm install feast-release feast-charts/feast \
+    --set global.registry.path=s3://feast/registries/prod \
+    --set global.project=<project name>
+```
+
+This chart will deploy two services: `feature-server` and `transformation-service`.
+Both must have read access to the registry file on cloud storage. Both will keep a copy of the registry in their memory and periodically refresh it.
+, so expect some delays in update propagation in exchange for better performance.
+
+#### Load balancing
+
+The next step would be to install an L7 Load Balancer (eg, [Envoy](https://www.envoyproxy.io/)) in front of the Java feature server.
+For seamless integration with Kubernetes (including services created by Feast Helm chart) we recommend using [Istio](https://istio.io/) as Envoy's orchestrator.
```

Comment:
Might be worth calling out that Istio also allows for an auth layer on top of Feast

Reply:
Istio allows a lot of stuff on top of Feast. I think having a link to their site is enough.